### PR TITLE
[Papercut][SW-16040] filterable properties on detail pages as links to filtered l…

### PR DESCRIPTION
**Pull request by dneustadt**
## Description

Please describe your pull request:
- Why is it necessary? filter articles by clicking on a property in product description
- What does it improve? properties of filterable property groups on detail page will link to the pre-filtered category with their respective filter values. the category in question is the one at the very end of the current breadcrumb.
- Does it have side effects? none so far

| Questions | Answers |
| --- | --- |
| BC breaks? | no |
| Tests pass? | yes |
| Related tickets? | https://issues.shopware.com/#/issues/SW-16040 |
| How to test? | detail/index on articles with associated properties |

### CODE REVIEW

File: engine/Shopware/Components/Compatibility/LegacyStructConverter.php [hunk 1]
```diff
@@ -803,6 +810,8 @@ public function convertPropertySetStruct(StoreFrontBundle\Struct\Property\Set $s
                 'groupName' => $set->getName(),
                 'value'     => implode(', ', $values),
                 'values'    => $values,
+                'isFilter'  => $group->isFilterable(),
```
Comment (OliverSkroblin) [inline engine/Shopware/Components/Compatibility/LegacyStructConverter.php]: rename `isFitler` to `isFilterable` please

File: engine/Shopware/Components/Compatibility/LegacyStructConverter.php [hunk 2]
```diff
@@ -786,6 +786,13 @@ public function convertPropertySetStruct(StoreFrontBundle\Struct\Property\Set $s
                 /**@var $option StoreFrontBundle\Struct\Property\Option */
                 $values[$option->getId()] = $option->getName();
             }
+            
+            $keyValues = array_map(
```
Comment (OliverSkroblin) [inline engine/Shopware/Components/Compatibility/LegacyStructConverter.php]: please use the `convertPropertyOptionStruct` function to convert this data.

`$keyValues = array_map([$this, 'convertPropertyOptionStruct'], $group->getOptions());`

File: themes/Frontend/Bare/frontend/detail/tabs/description.tpl [hunk 3]
```diff
@@ -43,7 +43,18 @@
 
 							{* Property content *}
 							{block name='frontend_detail_description_properties_content'}
-								<td class="product--properties-value">{$sProperty.value|escape}</td>
+								<td class="product--properties-value">
+                                    {if $sProperty.isFilter}
+                                        {$sBreadcrumbEnd = $sBreadcrumb|@end}
```
Comment (OliverSkroblin) [inline themes/Frontend/Bare/frontend/detail/tabs/description.tpl]: please use `$sArticle.categoryID` instead of `sBreadcrumb|@end`.

**Comment by OliverSkroblin** [comment]
Hi, 
the template changes are to big for a papercut. Please revert the template changes. The php changes can be merged (after comments solved), so that you can implement the template changes in your own local theme

File: engine/Shopware/Components/Compatibility/LegacyStructConverter.php [hunk 4]
```diff
@@ -803,6 +810,8 @@ public function convertPropertySetStruct(StoreFrontBundle\Struct\Property\Set $s
                 'groupName' => $set->getName(),
                 'value'     => implode(', ', $values),
                 'values'    => $values,
+                'isFilter'  => $group->isFilterable(),
+                'keyValues' => $keyValues,
```
Comment (OliverSkroblin) [inline engine/Shopware/Components/Compatibility/LegacyStructConverter.php]: I think `options` would be a better property name.

**Comment by dneustadt** [comment]
Thanks for the feedback. Done. I will offer the original template changes as a plugin instead. 💃

**Pull request merged**